### PR TITLE
Tokenization API support.

### PR DESCRIPTION
**Pull request by dmytrokhl**
## Description of changes
Support tokenization API

### CODE REVIEW

File: Sources/VGSCollectSDK/Core/Collector/VGSCollect+network.swift [hunk 1]
```diff
@@ -167,4 +167,25 @@ extension VGSCollect {
           block(response)
         }
     }
+  
+  public func tokenizeData(completion block: @escaping (VGSResponse) -> Void) {
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/Collector/VGSCollect+network.swift:171]: add docs for func

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSCVCTokenizationConfiguration.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+/// `VGSCVCTokenizationParameters` - parameters required for tokenization api.
+public struct VGSCVCTokenizationParameters: VGSTokenizationParametersProtocol {
+    
+    /// An Array of data classifiers.
+    public var classifiers: [String] = []
+  
+    /// Valut storage type.
+    public let storage: String = VGSVaultStorageType.VALOTILE.rawValue
+
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSCVCTokenizationConfiguration.swift:16]: Should be `VOLATILE`, not `VALOTILE `

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSCardHolderNameTokenizationConfiguration.swift [hunk 3]
```diff
@@ -0,0 +1,38 @@
+//
+//  VGSCardHolderNameTokenizationConfiguration.swift
+//  VGSCollectSDK
+//
+
+import Foundation
+
+/// `VGSSSNTokenizationParameters` - parameters required for tokenization api.
+public struct VGSCardHolderNameTokenizationParameters: VGSTokenizationParametersProtocol {
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSCardHolderNameTokenizationConfiguration.swift:9]: SSN config, typo?

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSExpDateTokenizationConfiguration.swift [hunk 4]
```diff
@@ -0,0 +1,37 @@
+//
+//  VGSExpDateTokenizationConfiguration.swift
+//
+
+import Foundation
+
+/// `VGSSSNTokenizationParameters` - parameters required for tokenization api.
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSExpDateTokenizationConfiguration.swift:7]: Fix typo in internal docs, not ssn

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSExpDateTokenizationConfiguration.swift [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    public let storage: String = VGSVaultStorageType.PERSISTANT.rawValue
+    
+    /// Data alies format.
+    public var format: String = VGSVaultAliasFormat.UUID.rawValue
+  
+    /// Defines if data should be tokenized.
+    public var shouldTokenize: Bool = true
+}
+
+/// `VGSSSNTokenizationConfiguration` - textfield configuration for textfield with type `.expDate`, required for work with tokenization api.
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSExpDateTokenizationConfiguration.swift:23]: Fix typo in internal docs, not ssn.

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+//  Created by Dmytro on 14.06.2022.
+//  Copyright © 2022 VGS. All rights reserved.
+//
+
+import Foundation
+
+
+public enum VGSVaultStorageType: String {
+  case PERSISTANT = "PERSISTANT"
+  case VALOTILE = "VALOTILE"
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift:14]: Fix typo, should be `VOLATILE`

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+//  VGSCollectSDK
+//
+//  Created by Dmytro on 14.06.2022.
+//  Copyright © 2022 VGS. All rights reserved.
+//
+
+import Foundation
+
+
+public enum VGSVaultStorageType: String {
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift:12]: add internal docs for public enum

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  case PERSISTANT = "PERSISTANT"
+  case VALOTILE = "VALOTILE"
+}
+
+public enum VGSVaultAliasFormat: String {
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift:17]: Add doc for public enum

File: demoapp/demoapp/UseCases/CardsDataTokenizationViewController.swift [hunk 9]
```diff
@@ -0,0 +1,188 @@
+//
+//  CardsDataTokenizationViewController.swift
+//  demoapp
+//
+
+import UIKit
+import VGSCollectSDK
+
+/// A class that demonstrates how to collect data from VGSTextFields and upload it to VGS
```
Comment (EugeneIOs) [inline demoapp/demoapp/UseCases/CardsDataTokenizationViewController.swift:9]: Fix copy paste text description for use case.

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSCardTokenizationConfiguration.swift [hunk 10]
```diff
@@ -0,0 +1,36 @@
+//
+//  VGSCardTokenizationConfiguration.swift
+//  VGSCollectSDK
+//
+
+import Foundation
+
+/// `VGSCardTokenizationParameters` - parameters required for tokenization api.
+public struct VGSCardTokenizationParameters: VGSTokenizationParametersProtocol {
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSCardTokenizationConfiguration.swift:9]: would suggest to use `VGSCardNumberTokenizationParameters` naming

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSCardTokenizationConfiguration.swift [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    
+    /// Data alies format.
+    public var format: String = VGSVaultAliasFormat.FPE_SIX_T_FOUR.rawValue
+  
+    /// Defines if data should be tokenized.
+    public let shouldTokenize: Bool = true
+}
+
+/// `VGSCardTokenizationConfiguration` - textfield configuration for textfield with type `.cardNumber`, required for work with tokenization api.
+public class VGSCardTokenizationConfiguration: VGSConfiguration, VGSTextFieldTokenizationConfigurationProtocol {
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSCardTokenizationConfiguration.swift:25]: would suggest to use `VGSCardNumberTokenizationConfiguration ` naming

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSCardHolderNameTokenizationConfiguration.swift [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+import Foundation
+
+/// `VGSCardHolderNameTokenizationParameters` - parameters required for tokenization api.
+public struct VGSCardHolderNameTokenizationParameters: VGSTokenizationParametersProtocol {
+    
+    /// An Array of data classifiers.
+    public var classifiers: [String] = []
+
+    /// Valut storage type.
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSCardHolderNameTokenizationConfiguration.swift:14]: Please fix typo, should be `Vault storage type`.

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSCardTokenizationConfiguration.swift [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+
+import Foundation
+
+/// `VGSCardTokenizationParameters` - parameters required for tokenization api.
+public struct VGSCardTokenizationParameters: VGSTokenizationParametersProtocol {
+    
+    /// An Array of data classifiers.
+    public var classifiers: [String] = []
+
+    /// Valut storage type.
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSCardTokenizationConfiguration.swift:14]: Please fix typo, should be `Vault storage type`.

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSExpDateTokenizationConfiguration.swift [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+
+import Foundation
+
+/// `VGSExpDateTokenizationParameters` - parameters required for tokenization api.
+public struct VGSExpDateTokenizationParameters: VGSTokenizationParametersProtocol {
+    
+    /// An Array of data classifiers.
+    public var classifiers: [String] = []
+
+    /// Valut storage type.
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSExpDateTokenizationConfiguration.swift:13]: Please fix typo, should be `Vault storage type`.

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSSSNTokenizationConfiguration.swift [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+
+import Foundation
+
+/// `VGSSSNTokenizationParameters` - parameters required for tokenization api.
+public struct VGSSSNTokenizationParameters: VGSTokenizationParametersProtocol {
+    
+    /// An Array of data classifiers.
+    public var classifiers: [String] = []
+
+    /// Valut storage type.
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSSSNTokenizationConfiguration.swift:15]: Please fix typo, should be `Vault storage type`.

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTokenizationConfiguration.swift [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+
+import Foundation
+
+/// `VGSTokenizationParameters` - parameters required for tokenization api.
+public struct VGSTokenizationParameters: VGSTokenizationParametersProtocol {
+    
+    /// An Array of data classifiers.
+    public var classifiers: [String] = []
+
+    /// Valut storage type.
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTokenizationConfiguration.swift:17]: Please fix typo, should be `Vault storage type`.

File: Sources/VGSCollectSDK/Core/Collector/VGSCollect+network.swift (hunk 1 above)
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/Collector/VGSCollect+network.swift:171]: `VGSResponse` or `VGSRequestResult`

File: Sources/VGSCollectSDK/Core/Collector/VGSCollect+fieldNameMapping.swift [hunk 17]
```diff
@@ -45,4 +45,20 @@ internal extension VGSCollect {
 			return collectFieldsJSON
 		}
 	}
+  
+  /// Map stored input for tokenizable textfields to JSON..
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/Collector/VGSCollect+fieldNameMapping.swift:49]: Remove extra `.`

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSCVCTokenizationConfiguration.swift [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+import Foundation
+
+/// `VGSCVCTokenizationParameters` - parameters required for tokenization api.
+public struct VGSCVCTokenizationParameters: VGSTokenizationParametersProtocol {
+    
+    /// An Array of data classifiers.
+    public var classifiers: [String] = []
+  
+    /// Valut storage type.
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSCVCTokenizationConfiguration.swift:14]: Should be `Vault`

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSExpDateTokenizationConfiguration.swift [hunk 19]
```diff
@@ -0,0 +1,37 @@
...
+    public var shouldTokenize: Bool = true
+}
+
+/// `VGSExpDateTokenizationConfiguration` - textfield configuration for textfield with type `.expDate`, required for work with tokenization api.
+public class VGSExpDateTokenizationConfiguration: VGSConfiguration, VGSTextFieldTokenizationConfigurationProtocol {
+  
+  /// `VGSExpDateTokenizationParameters` - tokenization configuration parameters.
+  public var tokenizationPatameters = VGSExpDateTokenizationParameters()
+
+  override public var type: FieldType {
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSExpDateTokenizationConfiguration.swift:29]: Please add `no doc` mark for public property.

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSSSNTokenizationConfiguration.swift [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+    public let shouldTokenize: Bool = true
+}
+
+/// `VGSSSNTokenizationConfiguration` - textfield configuration for textfield with type `.ssn`, required for work with tokenization api.
+public class VGSSSNTokenizationConfiguration: VGSConfiguration, VGSTextFieldTokenizationConfigurationProtocol {
+  
+  /// `VGSSSNTokenizationParameters` - tokenization configuration parameters.
+  public var tokenizationPatameters = VGSSSNTokenizationParameters()
+
+  override public var type: FieldType {
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSSSNTokenizationConfiguration.swift:31]: Please add `no doc` mark for public property.

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+  
+  /// VOLATILE data storage.
+  case VOLATILE = "VOLATILE"
+}
+
+/// Type of Alias format. Read more about avaliable formats: https://www.verygoodsecurity.com/docs/terminology/nomenclature#alias-formats .
+public enum VGSVaultAliasFormat: String {
+  case FPE_ACC_NUM_T_FOUR = "FPE_ACC_NUM_T_FOUR"
+  case FPE_ALPHANUMERIC_ACC_NUM_T_FOUR = "FPE_ALPHANUMERIC_ACC_NUM_T_FOUR"
+  case FPE_SIX_T_FOUR = "FPE_SIX_T_FOUR"
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift:24]: Please, add doc here.

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+  case FPE_SSN_T_FOUR = "FPE_SSN_T_FOUR"
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift:25]: Please, add doc here.

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+  case FPE_T_FOUR = "FPE_T_FOUR"
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift:26]: Please, add doc here.

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift [hunk 24]
```diff
@@ -0,0 +1,59 @@
...
+  case NUM_LENGTH_PRESERVING = "NUM_LENGTH_PRESERVING"
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift:27]: Please, add doc here.

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift [hunk 25]
```diff
@@ -0,0 +1,59 @@
...
+  case PFPT = "PFPT"
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift:28]: Please, add doc here.

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift [hunk 26]
```diff
@@ -0,0 +1,59 @@
...
+  case RAW_UUID = "RAW_UUID"
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift:29]: Please, add doc here.

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift [hunk 27]
```diff
@@ -0,0 +1,59 @@
...
+  
+  var tokenizationConfiguration: VGSTokenizationParametersProtocol { get }
+}
+
+/// Parameters describing textfield input tokenization.
+public protocol VGSTokenizationParametersProtocol {
+  /// Tokenization format
+  var format: String { get }
+  /// Storage type
+  var storage: String  { get }
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift:43]: Please add `.` at the end of the sentence.

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift [hunk 28]
```diff
@@ -0,0 +1,59 @@
...
+  /// Data classifiers
+  var classifiers: [String] { get }
+  /// Set if data should be tokenized. If `false` raw value will be stored in Vault
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSTextFieldTokenizationProtocol.swift:46]: Please add `.` at the end of the sentence.

File: Sources/VGSCollectSDK/Core/Collector/VGSCollect+network.swift (hunk 1 above)
Comment (dmytrokhl) [reply Sources/VGSCollectSDK/Core/Collector/VGSCollect+network.swift:171]: we don't have VGSRequestResult object in SDK.

File: Sources/VGSCollectSDK/Core/API/VGSResponse.swift [hunk 29]
```diff
@@ -31,3 +31,27 @@ import Foundation
...
+    /**
+     Success response case
+
+     - Parameters:
+        - code: response status code.
+        - body: response **JsonData** object.
+        - response: URLResponse object represents a URL load response.
+    */
+  case success(_ code: Int, _ body: JsonData?, _ response: URLResponse?)
+
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/API/VGSResponse.swift:46]: rename body to `jsonData`?

File: Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSExpDateTokenizationConfiguration.swift [hunk 30]
```diff
@@ -15,17 +15,29 @@ public struct VGSExpDateTokenizationParameters: VGSTokenizationParametersProtoco
...
+  
+  /// Input date format to convert.
+  public var inputDateFormat: VGSCardExpDateFormat?
+  
+  /// Output date format.
+  public var outputDateFormat: VGSCardExpDateFormat?
+  
+  // MARK: - VGSTokenizationParametersProtocol
+  /// `VGSExpDateTokenizationParameters` - tokenization configuration parameters.
+  public var tokenizationPatameters = VGSExpDateTokenizationParameters()
```
Comment (EugeneIOs) [inline Sources/VGSCollectSDK/Core/TokenizationConfiguration/VGSExpDateTokenizationConfiguration.swift:39]: Fix typo `tokenizationPatameters` to `tokenizationParameters`

**Pull request merged**